batchflow/models/tf/efficientnet.py

```diff
@@ -0,0 +1,252 @@
+                if parameter's value is a list, each element correspond to different stage
+
+                base : callable or list of callable
+                    Tensor processing function. Default is :meth:`.MobileNet_v2.block`
+
+                scalable : bool or list of bool
+                    indicates whether the block can be scaled
+
+                other : optional
+                    parameters for :meth:`.MobileNet_v2.block`
+
+    head : dict, optional
+        parameters for head (see :class:`EncoderDecoder`)
+        scalable : bool
+            indicates whether the block can be scaled
+
+    common : dict, optional
+        common parameters
+        width_factor, depth_factor : float, optional
+            scaling factors
+    """
+
+    resolution = 224  # image resolution used in original paper
+
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+
+        config['initial_block'] += dict(scalable=True, layout='cna', kernel_size=3, strides=2, filters=32)
+
+        config['body/encoder/num_stages'] = 7
+        config['body/encoder/order'] = ['block']
+        config['body/encoder/blocks'] += dict(base=MobileNet_v2.block,
+                                              repeats=[1, 2, 2, 3, 3, 4, 1],
+                                              scalable=True,
+                                              kernel_size=[3, 3, 5, 3, 5, 5, 3],
+                                              strides=[1, 2, 2, 2, 1, 2, 1],
+                                              filters=[16, 24, 40, 80, 112, 192, 320],
+                                              expansion_factor=[1, 6, 6, 6, 6, 6, 6],
+                                              se_block=dict(ratio=4))
+
+        config['body/encoder/downsample'] = None
+
+        config['body/embedding'] = None
+        config['body/decoder'] = None
+
+        config['head'] += dict(scalable=True, layout='cna V df', kernel_size=1, strides=1, filters=1280,
+                               dropout_rate=0.2)
+
+        config['common'] += dict(activation=tf.nn.swish, width_factor=1.0, depth_factor=1.0)
+
+        config['loss'] = 'ce'
+
+        return config
+
+    def build_config(self, names=None):
+        config = super().build_config(names)
+
+        if config.get('head/units') is None:
+            config['head/units'] = self.num_classes('targets')
+
+        w_factor = config.get('common/width_factor')
+        d_factor = config.get('common/depth_factor')
+
+        for path in ('initial_block/', 'body/encoder/blocks/', 'head/'):
+            scalable = config.get(path + 'scalable')
+            if scalable:
+                for param, factor, fun in [('filters', w_factor, self.round_filters),
+                                           ('repeats', d_factor, self.round_repeats)]:
+                    if factor != 1:
+                        val = config.get(path + param)
+                        if val:
```

Comment:
why?

Reply:
In most cases `factor` is not equal to one, so it would result in less checks on average

Reply:
Really not sure about "most cases" and what the average checks number would be... It will depend on use cases. This piece of code is definitely not a bottleneck so I would prefer to leave it as is